builder/parallels/common/step_attach_cd.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+)
+
+// StepAttachCD is a step that attaches a cd to the virtual machine.
+//
+// Uses:
+//
+//	driver Driver
+//	ui packersdk.Ui
+//	vmName string
+//
+// Produces:
+type StepAttachCD struct {
+	cdRomType   string `mapstructure:"cdrom_type"`
+	cdPath      string
+	cdromDevice string
+}
+
+// Run adds a virtual CD to the VM and attaches the image.
+// If the image is not specified, then this step will be skipped.
+func (s *StepAttachCD) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	// Determine if we even have a cd to attach
+	var cdPath string
+	if cdPathRaw, ok := state.GetOk("cd_path"); ok {
+		cdPath = cdPathRaw.(string)
+	} else {
+		log.Println("No cd found, not attaching.")
+		return multistep.ActionContinue
+	}
+
+	driver := state.Get("driver").(Driver)
+	ui := state.Get("ui").(packersdk.Ui)
+	vmName := state.Get("vmName").(string)
+
+	ui.Say("Attaching cd...")
+	//Use CLI instead of driver wrapper so we can control iface
```

Comment:
This creates duplicate code.

Please use the driver to add cdrom, and don't provide a '--iface' as of now, sata is default. Use the returning name to delete it.

Reply:
Technically I am using the driver, just to call the CLI :)

I chose to use the CLI because the driver doesn't let you set the type. `scsi` and less so `ide` could be useful to support in the future and this allows it with minimal fuss.

Let me know if you don't care about `scsi` or `ide support` and I can change it.

Reply:
Yes, we don't need to care about `scsi` or `ide` as of now. We can add interface support to driver itself in future if necessary.

Reply:
Ok, it's now done via the driver.